src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -733,10 +733,10 @@ public new CSharpCompilation RemoveSyntaxTrees(IEnumerable<SyntaxTree> trees)
                     var loadedSyntaxTreeMap = syntaxAndDeclarations.GetLazyState().LoadedSyntaxTreeMap;
                     if (SyntaxAndDeclarationManager.IsLoadedSyntaxTree(tree, loadedSyntaxTreeMap))
                     {
-                        throw new ArgumentException(string.Format(CSharpResources.SyntaxTreeFromLoadNoRemoveReplace, tree), $"{nameof(trees)}[{i}]");
+                        throw new ArgumentException(string.Format(CSharpResources.SyntaxTreeFromLoadNoRemoveReplace, tree.FilePath), $"{nameof(trees)}[{i}]");
                     }
 
-                    throw new ArgumentException(string.Format(CSharpResources.SyntaxTreeNotFoundTo, tree), $"{nameof(trees)}[{i}]");
+                    throw new ArgumentException(string.Format(CSharpResources.SyntaxTreeNotFoundToRemove, tree.FilePath), $"{nameof(trees)}[{i}]");
```

Comment:
FilePath may be null (or at least empty).  It's a completely optional part of the whole API.

Reply:
Right, but do you have a better way of identifying the syntax tree for diagnostic purposes?

Considering that the whole syntax tree can be hundreds of lines long, I think that's a much worse option. No information or ambiguous information still seems better to me than overwhelmingly huge amount of information for an exception message.

Reply:
It could potentially be referred to by index.

Reply:
What index? If you're talking about index within the current compilation, then I think that could make sense in other situations, but not here, where the exception is that the `SyntaxTree` is not contained in this compilation.

Reply:
looking at the code, it should just be what it was before.  i.e. ```$"{nameof(trees)}[{i}]"```.  So it's telling you the index in the argument list where the problem tree occurs.

Reply:
Ah, I understand what index you mean now. You're right that that's potentially useful information, but that's still passed as `paramName` to the exception (which means it will show up in its `Message` property), I didn't change that.

Are you saying that the `message` parameter should be just "SyntaxTree not found to remove", which would make the `Message` property e.g.:

> SyntaxTree not found to remove
> Parameter name: trees[4]

And how would this apply to the other cases I changed in this PR, where there is no index (since there is just one `SyntaxTree`)?

Reply:
> Are you saying that the message parameter should be just "SyntaxTree not found to remove", 

That, or> SyntaxTree 'trees[4]' not found to remove"

Reply:
> And how would this apply to the other cases I changed in this PR, where there is no index 

If it's just a single tree, then no need to print this information as it will be obvious from teh call.

Reply:
@CyrusNajmabadi Done. I thought there's no need to repeat `paramName` in the `message`, so I went with the simpler version.